### ISSUE TITLE
FtpClient.AutoConnect() does not work after upgrading from 34.0.0 to 37.0.2 (fixed) and Session Resume failure

### ISSUE THREAD

**Issue by barbaraacsilva** [robinrodricks/FluentFTP#855]
**FTP:** https://dlptest.com/ftp-test/

**Computer OS:** Windows

**FluentFTP Version:** 37.0.2

I'm having issues connecting to a specific FTP server with AutoConnect(), and I noticed my FluentFTP version (34.0.0) could be updated to 37.0.2, so I tried that. I haven't even been able to try and connect again to the specific server, because while debugging locally my application stopped connecting to the public FTP test site from DLP, which used to work just fine with AutoConnect().

**Logs:**

```

# AutoConnect()
# AutoDetect(True, False)
# Connect()
Status:   Connecting to ***:21
Response: 220 Welcome to the DLP Test FTP Server
Command:  AUTH TLS
Response: 234 Proceed with negotiation.
Command:  QUIT
Warning:  FtpClient.Disconnect(): Exception caught and discarded while closing control connection: System.Security.Authentication.AuthenticationException: The remote certificate was rejected by the provided RemoteCertificateValidationCallback.
   at System.Net.Security.SslStream.<ThrowIfExceptional>g__ThrowExceptional|137_0(ExceptionDispatchInfo e)
   at System.Net.Security.SslStream.Write(Byte[] buffer, Int32 offset, Int32 count)
   at FluentFTP.FtpSocketStream.Write(Byte[] buffer, Int32 offset, Int32 count)
   at FluentFTP.FtpSocketStream.WriteLine(Encoding encoding, String buf)
   at FluentFTP.FtpClient.Execute(String command)
   at FluentFTP.FtpClient.Disconnect()
Status:   Disposing FtpSocketStream...
# Connect()
Status:   Connecting to ***:21
Status:   There is stale data on the socket, maybe our connection timed out or you did not call GetReply(). Re-connecting...
Status:   Disposing FtpSocketStream...
Status:   Not sending QUIT because the connection has already been closed.
Status:   Disposing FtpSocketStream...

```

**Updates**
AutoConnect() worked when called after ValidateAnyCertificate = true (I tried it after analyzing the previous exception message).
But now I'm getting an exception when calling GetListing(path):

```
# GetListing("/test1/test2/test3", Auto)
Command:  TYPE I
Response: 200 Switching to Binary mode.
# OpenPassiveDataStream(PASV, "LIST /test1/test2/test3", 0)
Command:  PASV
Response: 227 Entering Passive Mode (35,163,228,146,4,21).
Status:   Connecting to ***:1045
Command:  LIST /test1/test2/test3
Response: 150 Here comes the directory listing.
Status:   FTPS Authentication Successful
Status:   Time to activate encryption: 0h 0m 0s.  Total Seconds: 0,4922175.
+---------------------------------------+
-----------------------------------------
Status:   Disposing FtpSocketStream...
# CloseDataStream()
Response: 522 SSL connection failed; session reuse required: see require_ssl_reuse option in vsftpd.conf man page
Status:   Disposing FtpSocketStream...
```

**Update - March 17**
I was able to connect successfully using this code:

```
FtpProfile autoConnectProfile = null;

try
{
    autoConnectProfile = ftpClient.AutoConnect();
}
catch (Exception)
{
    // nothing
}

if (autoConnectProfile == null)
{
    var manualFtpProfile = new FtpProfile
    {
        Host = ftpClient.Host,
        Credentials = ftpClient.Credentials,
        Encoding = Encoding.UTF8,
        Protocols = SslProtocols.None,
        DataConnection = FtpDataConnectionType.PASV,
        Encryption = FtpEncryptionMode.None,
        Timeout = 30000
    };

    ftpClient.Connect(manualFtpProfile);
}
```

**Comment by FanDjango** [comment]
I have followed up on this issue, and here are the results:

With FluentFTP V42+:

1. `AutoConnect()` to the above mentioned FTP test server, `ftp://ftp.dlptest.com/` works fine. So that bug has been fixed in the meantime.

**but** in the further text of your issue:

2. `GetListing()` produces `522 SSL connection failed; session reuse required: see require_ssl_reuse option in vsftpd.conf man page`. This tells us: It is a `vsftpd` server that requires `SSL Session Resume`.

Although we support SSL Session Resume in TLS 1.2 connections, **there seem to be some specific versions of vsftpf ( unknown ) and a specific versions of FileZilla-Server ( = 0.9.60 ) where SSL Session Resume does not work**.

Sadly, the DLP FTP Test server has remove the standard vsftpd version text from his welcome message, so we still do not know which version(s) of vsftpd exhibit this problem.

**and** your March 17 Update:

By setting `Encryption = FtpEncryptionMode.None`, you have turned off **encryption**, thereby using an insecure connect, since there is no SSL, there is no SSL Session Resume and thus no failure. But surely that is not what you intended?
